src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (typeof this === 'string') {
+      this.split(' ');
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += '';
```

Comment:
You can remove this line as it's unnecessary to append an empty string. The else block is sufficient to handle non-null and non-undefined values.

Reply:
Consider fixing the previous comment.